spotbugs/src/main/java/edu/umd/cs/findbugs/util/SecurityManagerHandler.java

```diff
@@ -0,0 +1,60 @@
+package edu.umd.cs.findbugs.util;
+
+/**
+ * Since Java 17, the security manager is deprecated for removal and invoking related methods
+ * causes a warning to be printed to the console. This intermediate disables the use of
+ * security manager-related APIs on Java 17 or later, unless using the security manager is
+ * explicitly configured by setting the <i>edu.umd.cs.findbugs.securityManagerDisabled</i>
+ * property.
+ */
+public class SecurityManagerHandler {
+
+    /**
+     * Determines if the security manager is used by SpotBugs.
+     */
+    public static boolean SECURITY_MANAGER_DISABLED;
+
+    static {
+        boolean securityManagerDisabled;
+        try {
+            String property = System.getProperty("edu.umd.cs.findbugs.securityManagerDisabled");
+            if (property != null) {
+                securityManagerDisabled = Boolean.parseBoolean(property);
+            } else {
+                String version = System.getProperty("java.version");
+                if (version.startsWith("1.")) {
+                    version = version.substring(2, 3);
+                } else {
+                    int index = version.indexOf(".");
+                    if (index != -1) {
+                        version = version.substring(0, index);
+                    }
+                }
+                securityManagerDisabled = Integer.parseInt(version) > 16;
+            }
+        } catch (Throwable ignored) {
+            securityManagerDisabled = false;
```

Comment:
It's also nice to have a debug log here for debugging purpose:

```suggestion
            LoggerFactory.getLogger(SecurityManagerHandler.class)
                .debug("failed to detect the ability of security manager feature, so treat it as available", e);
            securityManagerDisabled = false;
```

Reply:
I added some logging.